mlflow/models/model.py

```diff
@@ -344,7 +344,7 @@ def model_id(self) -> str:
 
         :getter: Gets the model ID of the logged model
         """
-        return self._logged_model.model_id
+        return self._logged_model and self._logged_model.model_id
```

Comment:
` self._logged_model.model_id` throws an `AttributeError` if ` self._logged_model` is `None`.

Reply:
I would prefer `self._logged_model.model_id if self._logged_model else None`, this `and` operation is a bit implicit

Reply:
sounds good!